python/ray/data/_internal/planner/plan_read_op.py

```diff
@@ -14,13 +14,18 @@
     MapTransformFn,
 )
 from ray.data._internal.logical.operators.read_operator import Read
-from ray.data._internal.util import _warn_on_high_parallelism
+from ray.data._internal.util import _warn_on_high_parallelism, call_with_retry
 from ray.data.block import Block
 from ray.data.context import DataContext
 from ray.data.datasource.datasource import ReadTask
 
 TASK_SIZE_WARN_THRESHOLD_BYTES = 100000
 
+# Transient errors that can occur during longer reads. Trigger retry when these occur.
+READ_FILE_RETRY_ON_ERRORS = ["AWS Error NETWORK_CONNECTION", "AWS Error ACCESS_DENIED"]
+READ_FILE_MAX_ATTEMPTS = 10
+READ_FILE_RETRY_MAX_BACKOFF_SECONDS = 32
```

Comment:
This has been repeated multiple times in the code base. can we consolidate them with a util function?

Reply:
@raulchen i intended to keep these constants separate from the retry errors for opening files, e.g. https://github.com/ray-project/ray/blob/bfc1f78e7a4e451958f5791e29c8250980f8295a/python/ray/data/datasource/file_based_datasource.py#L67-L74

i'm using the common util function `call_with_retry` with these constants. do you mean consolidate these sets of constants into their own file?

Reply:
ok. might be better to also define those constants in a unified place. but not a big deal.

Reply:
i couldn't find a centralized `constants.py` or other similar file for ray data constants. should we expose these as parameters from `DataContext`?

Reply:
okay to keep it the current way. let's refactor later if needed.

Reply:
I just realized that we already have `write_file_retry_on_errors` in DataContext`. Let's also follow this pattern?